tools/2025-01-gitops-technical-design.md

```diff
@@ -0,0 +1,435 @@
+* If the feature shares the same objectives of the existing design, link
+  to the existing doc and section rather than repeat the same context.
+* If the feature has a scenario, UX, or other product feature design doc,
+  link it here and summarize the important parts.
+-->
+
+> **Issue Reference:** #6689
+
+### Goals
+
+<!--
+Describe goals to define why we are doing this work, how we will make
+priority decisions, and how we will determine success.
+-->
+
+- Enable seamless integration of GitOps workflows with Radius + Bicep. Users should be able to specify their Bicep files and parameters in a Git repository and have the resources within them automatically deployed to their Kubernetes cluster.
+- Ensure that the infrastructure defined in the Git repository is accurately reflected in the Kubernetes cluster. This includes creating, updating, and deleting resources as necessary, based on the changes in the Git repository.
+- Provide an experience that is consistent with existing Flux workflows. Users should be able to use Flux as they do today, with the addition of the Radius Flux Controller for managing Bicep files.
+
+### Non goals
+
+<!--
+Describe non-goals to identify something that we won’t be focusing on
+immediately. We won’t be expending any effort on these matters. If there
+will be follow-ups after this work, list them here. If there are things
+we plan to do in the future, but are out of scope of this design, list
+them here. Provide a brief explanation on why this is a non-goal.
+-->
+
+- Supporting non-Flux GitOps tools (e.g., ArgoCD).
+- Supporting non-Bicep IaC formats.
+- Radius Dashboard support
+
+### User scenarios (optional)
+
+<!--
+Describe the user scenarios for this design. Ensure that you define the
+roles and personas in these user scenarios when it requires API design.
+If you have an existing issue that describes the user scenarios, please
+link to that issue instead.
+-->
+
+#### Helly can use Radius and Bicep to simplify application and infrastructure management
+
+Helly is an infrastructure operator for an enterprise company. Her team manages a production application that is deployed on Kubernetes and uses AWS resources. To simplify the management of the application and infrastructure, Helly decides to use Radius and Bicep. She writes Bicep manifests that define the AWS resources required by the application, such as S3 buckets and RDS databases, as well as Radius resources that define container applications. Helly then uploads these Bicep files to a Git repository. She then writes a Flux configuration which includes Flux installation, Radius installation, and Flux setup for tracking her Git repository. Finally, Helly applies the Flux configuration to her Kubernetes cluster, and Radius deploys the resources defined in the Bicep manifests to the cluster. Now, any changes to the Bicep files in the Git repository are automatically reflected in the Kubernetes cluster, enabling a seamless GitOps workflow.
```

Comment:
Bootstrapping in which Flux deploys Radius, which then receives notifications from Flux to deploy Bicep: is this a supported scenario?

Reply:
yes, this is already supported because Radius can be installed by a Helm chart